### PR TITLE
[backport] Fix SCE with ChainerX and normalize

### PR DESCRIPTION
**Pull request by takagi**
Backport of #8301

### CODE REVIEW

**Comment by takagi** [comment]
[automatic post] Jenkins, test this please.

**Comment by takagi** [comment]
Please follow #8307 to be merged.

**Comment by takagi** [comment]
Now ##8307 is merged and I've update this PR.

File: chainer/functions/loss/softmax_cross_entropy.py [hunk 1]
```diff
@@ -111,15 +109,22 @@ def _is_chainerx_supported(self, input_arrays):
...
-        # TODO(beam2d): implement mean
-        y = -(score * mask).sum() * (1 / x.shape[0])
+
+        neg_log_p = -score * mask
+        if self.reduce == 'mean':
+            if self.normalize:
+                count = mask.sum()
+            else:
+                count = x.shape[0]
+            y = neg_log_p.sum() * (1 / count)
```
Comment (hvy) [inline chainer/functions/loss/softmax_cross_entropy.py]: Could you fix the test failure on this line? It's due to the fact that https://github.com/chainer/chainer/pull/7075 was never backported. As `count` doesn't participate in the graph, you could just in the `if self.normalize:` path convert the `count` to a Python primitive number. Changing the `*` with `/` might be fine as well.
```
TypeError: unsupported operand type(s) for /: 'int' and 'chainerx.ndarray'
```

Comment (takagi) [reply chainer/functions/loss/softmax_cross_entropy.py]: Fixed!

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
Hm, seems like CI is failing.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by takagi** [comment]
Waiting for https://github.com/chainer/chainer/pull/8075#issuecomment-548304889 to pass the CI failure.

**Comment by takagi** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
Some CI failures that are unrelated. Thanks for the changed, LGTM.

**Pull request merged**